AdminCore.ScheduledJobs/ScheduleRunner.cs

```diff
@@ -0,0 +1,73 @@
+﻿using AdminCore.Common.Interfaces;
+using AdminCore.DAL;
+using AdminCore.DTOs;
+using Quartz;
+using Quartz.Impl;
+using System;
+
+namespace AdminCore.ScheduledJobs
+{
+  public class ScheduleRunner : IScheduleRunner, IDisposable
+  {
+    private readonly IScheduler _scheduler;
+
+    public ScheduleRunner(IDatabaseContext databaseContext)
+    {
+      var factory = new StdSchedulerFactory();
+      _scheduler = factory.GetScheduler().Result;
+      _scheduler.Start();
+
+      _scheduler.JobFactory = new DatabaseJobFactory(databaseContext);
+    }
+
+    public void Dispose()
+    {
+      _scheduler.Shutdown(true);
+      GC.SuppressFinalize(this);
+    }
+
+    public void StartSchedule(ScheduleDto schedule)
+    {
+      if (schedule.IsActive)
+      {
+        var trigger = GetTriggerForScheduler(schedule);
+        var job = GetJob(schedule);
+        _scheduler.ScheduleJob(job, trigger);
+      }
+    }
+
+    private static IJobDetail GetJob(ScheduleDto schedule)
+    {
+      return schedule.IsActive ? GetJobForScheduler<StartupJob>(schedule) : GetJobForScheduler<ShutdownJob>(schedule);
```

Comment:
Don’t need shutdown and can we rename startup to PruneExpiredEventsJob.

Can we also have a jobs enum and switch on it? Just cleaner to maintain.